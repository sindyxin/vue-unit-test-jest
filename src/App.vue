<template>
  <div id="app">
    <MessageList :messages="messages">
      <header slot="header">
        Awesome header
      </header>
      <Message v-for="message in messages" :key="message" :message="message" @message-clicked="handleMessageClick"></Message>
    </MessageList>
    <Form></Form>
  </div>
</template>

<script>
import MessageList from "@/components/MessageList";
import Message from "@/components/Message";
import Form from "@/components/Form";

export default {
  name: "app",
  props:['message'],
  data: () => ({ messages: ["Hey John", "Howdy Paco"] }),
  components: {
    MessageList,
    Message,
    Form
  },
  methods: {
    handleMessageClick(message) {
      console.log(message);
    }
  }
};
</script>


