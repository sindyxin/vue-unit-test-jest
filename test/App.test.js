import { shallowMount } from '@vue/test-utils'
// import App from '../src/App'
import App from '@/App'

describe('App.test.js', () => {
    let cmp

    beforeEach(() => {
        cmp = shallowMount(App)
        cmp.setData({ messages: ['Cat'] })
    })

    it('equals messages to ["Cat"]', () => {
        // Within cmp.vm, we can access all Vue instance methods
        expect(cmp.vm.messages).toEqual(['Cat'])
    })

    it('has the expected html structure', () => {
        expect(cmp.element).toMatchSnapshot()
    })
})

/**
 * Without vue-test-utils
 */
// import Vue from 'vue'
// import App from '../src/App'

// describe('App.test.js', () => {
//   let cmp, vm

//   beforeEach(() => {
//     cmp = Vue.extend(App)
//     vm = new cmp({
//       data: {
//         messages: ['Cat']
//       }
//     }).$mount()
//   })

//   it('equals messages to ["Cat"]', () => {
//     expect(vm.messages).toEqual(['Cat'])
//   })

//   it('has the expected html structure', () => {
//     expect(vm.$el).toMatchSnapshot()
//   })
// })


//     "webpack": "^4.19.1",
// "webpack-bundle-analyzer": "^3.0.2",
// "webpack-cli": "^3.1.0",
// "webpack-dev-server": "^2.9.7",
// "webpack-merge": "^4.1.0"
// "extract-text-webpack-plugin": "^3.0.0",