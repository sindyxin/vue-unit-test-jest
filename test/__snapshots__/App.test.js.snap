// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.test.js has the expected html structure 1`] = `
<div
  id="app"
>
  <messagelist-stub
    messages="Cat"
  >
    <header>
      
      Awesome header
    
    </header>
     
    <message-stub
      author="Paco"
      message="Cat"
    />
  </messagelist-stub>
   
  <form-stub />
</div>
`;
